db/view/view_update_generator.cc

```diff
@@ -39,10 +39,25 @@ future<> view_update_generator::start() {
                 if (result == stop_iteration::yes) {
                     break;
                 }
-                entry.t->move_sstable_from_staging_in_thread(entry.sst);
+                // collect all staging sstables to move in a map, grouped by table.
+                auto t = entry.t.get();
+                auto it = _sstables_to_move.find(t);
+                if (it == _sstables_to_move.end()) {
+                    _sstables_to_move.emplace(t, entry);
+                } else {
+                    it->second.sstables.push_back(entry.sst);
+                }
                 _registration_sem.signal();
```

Comment:
If the `push_back()` above throws this will be leaked.

Reply:
That was true also before the change, for move_sstable_from_staging_in_thread (and to read_rows_flat and consume_in_thread), no?

Reply:
Yes, there is preexisting potentially throwing code above. We still probably should fix this although not in this patch. My gut feeling is that not releasing this semaphore will convert some exception (`std::bad_alloc`) into some deadlock.